### PR TITLE
Userlog Service Improvements

### PR DESCRIPTION
**Pull request by kobergj**
Improves userlog service with proper messages and enhances API

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>cs3ApiTests-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/19671/42/1) failed. Further test are cancelled...

File: go.sum [hunk 1]
```diff
@@ -345,8 +345,8 @@ github.com/crewjam/httperr v0.2.0/go.mod h1:Jlz+Sg/XqBQhyMjdDiC+GNNRzZTD7x39Gu3p
 github.com/crewjam/saml v0.4.6/go.mod h1:ZBOXnNPFzB3CgOkRm7Nd6IVdkG+l/wF+0ZXLqD96t1A=
 github.com/crewjam/saml v0.4.9 h1:X2jDv4dv3IvfT9t+RhADavzNFAcq3fVxzTCIH3G605U=
 github.com/crewjam/saml v0.4.9/go.mod h1:9Zh6dWPtB3MSzTRt8fIFH60Z351QQ+s7hCU3J/tTlA4=
-github.com/cs3org/reva/v2 v2.12.1-0.20230306101206-3de7ec137baf h1:Qj8tUBW89yglwUx+q2vZN1mIlkrHdd/FVMAN6Wg4T0U=
-github.com/cs3org/reva/v2 v2.12.1-0.20230306101206-3de7ec137baf/go.mod h1:EGPAVHVqMHZ6GBo4T6QMw13D0djGU8ZCi/o5siJ9IUs=
+github.com/cs3org/reva/v2 v2.12.1-0.20230308104654-4ebb727fa197 h1:CGUH/W7ZNN3pm7ElRq3vYfVIkoHj3975EY7veaR/nyY=
```
[the remark this answers is not in the document]
Comment (kobergj) [reply go.sum:348]: Not current, but still more up-to-date then master :wink: It is pointing to this commit: https://github.com/cs3org/reva/commit/4ebb727fa19723bfff67b9eb323f49bce79a47a8

**Review by micbar — APPROVED** [review]
small nitpicks

File: services/userlog/pkg/service/service.go [hunk 2]
```diff
@@ -378,49 +315,99 @@ func (ul *UserlogService) gatherSpaceMembers(ctx context.Context, space *storage
...
+func (ul *UserlogService) getSpace(ctx context.Context, spaceID string) (*storageprovider.StorageSpace, error) {
+	res, err := ul.gwClient.ListStorageSpaces(ctx, listStorageSpaceRequest(spaceID))
 	if err != nil {
 		return nil, err
 	}
 
-	if r.GetStatus().GetCode() != rpc.Code_CODE_OK {
-		return nil, fmt.Errorf("unexpected status code from gateway client: %d", r.GetStatus().GetCode())
+	if res.GetStatus().GetCode() != rpc.Code_CODE_OK {
+		return nil, fmt.Errorf("Unexpected status code while getting space: %v", res.GetStatus().GetCode())
```
Comment (butonic) [inline services/userlog/pkg/service/service.go:362]: Maybe build the error from code and message?
```suggestion
		return nil, fmt.Errorf("Error while getting space: (%v) %s", res.GetStatus().GetCode(), res.GetStatus().GetMessage())
```

**Review by C0rby — APPROVED** [review]

File: services/userlog/pkg/service/service.go (hunk 2 above)
Comment (kobergj) [reply services/userlog/pkg/service/service.go:362]: OK. I'll fix it here: https://github.com/owncloud/ocis/pull/5753

**Pull request merged**